### PR TITLE
Change requirements to Cirq 0.10

### LINKED ISSUE #716: Consider reverting cirq-google until deprecation warnings will disappear.
When we passed from `cirq` to `cirq-core`, a lot of deprecation warning appear each time we import something.
E.g. if a user tries to run:

```
python
>>> from mitiq.zne import execute_with_zne
```

The following long message appears:

<details>
--- Logging error ---
Traceback (most recent call last):
  File "/home/andrea/anaconda3/lib/python3.7/site-packages/cirq/__init__.py", line 581, in <module>
    create_attribute=True,
  File "/home/andrea/anaconda3/lib/python3.7/site-packages/cirq/_compat.py", line 546, in deprecated_submodule
    new_module = importlib.import_module(new_module_name)
  File "/home/andrea/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'cirq_google'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/andrea/anaconda3/lib/python3.7/logging/__init__.py", line 1025, in emit
    msg = self.format(record)
  File "/home/andrea/anaconda3/lib/python3.7/logging/__init__.py", line 869, in format
    return fmt.format(record)
  File "/home/andrea/anaconda3/lib/python3.7/logging/__init__.py", line 608, in format
    record.message = record.getMessage()
  File "/home/andrea/anaconda3/lib/python3.7/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "<stdin>", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/andrea/data/mitiq/mitiq/__init__.py", line 16, in <module>
    from mitiq._about import about
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/andrea/data/mitiq/mitiq/_about.py", line 23, in <module>
    from cirq import __version__ as cirq_version
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/andrea/anaconda3/lib/python3.7/site-packages/cirq/__init__.py", line 585, in <module>
    warning("Can't import cirq.google: ", ex)
Message: "Can't import cirq.google: "
Arguments: (ModuleNotFoundError("No module named 'cirq_google'"),)
</details>

The same problem also appears in some examples in the docs. See e.g.:
https://mitiq.readthedocs.io/en/latest/examples/simple_landscape.html
https://mitiq.readthedocs.io/en/latest/examples/maxcut-demo.html#define-the-qaoa-cost-hamiltonian-using-cirq

** Proposal **
Temporarily require the full `cirq` package, until the warning disappear?
They will probably fix this at some point: https://github.com/quantumlib/Cirq/issues/4106

### PR DESCRIPTION
**Pull request by andreamari**
Fixes #716 by adding `cirq` in `requirements.txt.`
In this way, the existing warnings in the docs should disappear.

### CODE REVIEW

**Review by nathanshammah — APPROVED** [review]

**Comment by nathanshammah** [comment]
Actually the documentation tests fail with this message
```
Exception occurred:
  File "/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/sphinx/ext/autodoc/mock.py", line 151, in mock
    sys.meta_path.remove(finder)
ValueError: list.remove(x): x not in list
The full traceback has been saved in /tmp/sphinx-err-uu6ygqlb.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make[1]: *** [Makefile:20: html] Error 2
make[1]: Leaving directory '/home/runner/work/mitiq/mitiq/docs'
make: *** [Makefile:34: docs] Error 2
Error: Process completed with exit code 2.
```

**Comment by rmlarose** [comment]
It is related. Sphinx doesn't like the way cirq_google is disguised as cirq.google. Sorry I forgot about this.

So you'd have to add cirq to requirements.txt instead of this @andreamari, unless you want to try and fix the present situation.

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 numpy~=1.20.1
 scipy~=1.4.1
-cirq-core~=0.11.0
+cirq~=0.10.0
```
Comment (rmlarose) [inline requirements.txt:3]: Why 0.10?

Comment (rmlarose) [reply requirements.txt:3]: Oh, I see "just to test" in the commit history :-)

Comment (rmlarose) [inline requirements.txt:3]:
```suggestion
cirq~=0.11.0
```

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 numpy~=1.20.1
 scipy~=1.4.1
-cirq-core~=0.11.0
+cirq~=0.11.0
```
Comment (rmlarose) [inline requirements.txt:3]:
```suggestion
cirq~=0.10.0
```

**Comment by nathanshammah** [comment]
No issues any more, indeed, @andreamari!

**Comment by rmlarose** [comment]
For posterity:

Main reason for this is to get rid of "Warning: `cirq_google` not installed" messages in the documentation. We cannot simply install `cirq_google`, unfortunately, because the module renaming to `cirq.google` seems to not play well with sphinx - see https://github.com/unitaryfund/mitiq/pull/717#issuecomment-858584681. The easiest, though not ideal, solution is thus moving back to Cirq 0.10 for the moment, as we do not rely on any new features in Cirq 0.11.

**Pull request merged**